### DISCUSSION TITLE
Support aarch64-linux

### DISCUSSION THREAD

**Discussion by agatekartik** [SeaQL/FireDBG.for.Rust#29] [Bug]
Here is the output of the run:

> curl https://raw.githubusercontent.com/SeaQL/FireDBG.for.Rust/main/install.sh -sSf | sh
> info: downloading FireDBG from `https://github.com/SeaQL/FireDBG.for.Rust/releases/download/1.74.2/aarch64**-debian12**.tar.gz`
> curl: (22) The requested URL returned error: 404
> FireDBG: failed to download https://github.com/SeaQL/FireDBG.for.Rust/releases/download/1.74.2/aarch64**-debian12**.tar.gz
> FireDBG: this may be a standard network error, but it may also indicate
> FireDBG: that FireDBG's release process is not working. When in doubt
> FireDBG: please feel free to open an issue!
> 

There's aarch64-darwin but no aarch64-debian12

**Comment by tyt2y3** [comment]
Yeah sorry, we ship prebuilt binaries for Linux on x64 only. I think it can be built from source, but I actually don't have a machine to test it on.

**Reply by agatekartik** [reply]
We can test on one of my Raspberry Pi 4. Let me know steps/modalities.

**Reply by tyt2y3** [reply]
Ah, I remember having a Pi somewhere in our storage, I hope it is still there. You need to download the `codelldb-aarch64-linux.vsix
` binaries then put them in the right path https://github.com/SeaQL/FireDBG.for.Rust/blob/main/BUILDING.md
The dependencies for Debian are https://github.com/SeaQL/FireDBG.for.Rust/blob/main/build/Dockerfile.debian12

Then you can go into the `debugger` directory and run the tests:

```sh
cd FireDBG.for.Rust/debugger
cargo b
cargo t --test '*'
```

If everything passes, then it's good news! I think we've had a few nuisances porting from linux to darwin on x64.

**Comment by agatekartik** [comment]
Hi - I ran the tests on my raspberry Pi 4 after modifying the Dockerfile.debian12 to point to aarch64 and installing lldb for aarch64 and following the detailed instructions.

The tests passed :-)

Kindly let me know if there are any additional tests I should run and/or any artefacts I need to share for the test run and results

**Reply by tyt2y3** [reply]
oh wow! thank you for the good news. may be we can release a local installation guide

**Reply by tyt2y3** [reply]
If all the tests passed, you can install them by:

```sh
cd command
cargo install --path . --debug --force

cd ../debugger
cargo install --path . --debug --force

cd ../indexer
cargo install --path . --debug --force
```

You still have to symlink or copy the dependencies to `$CARGO_HOME/firedbg-lib`. i.e.

```sh
~/.cargo/bin/firedbg-lib$ ls bin
lldb  lldb-argdumper  lldb-server
~/.cargo/bin/firedbg-lib$ ls lib
liblldb.so  libpython39.so  lldb-python  python3.9
```

**Reply by agatekartik** [reply]
Thanks - if there are any artefacts needed to demonstrate the passing tests, let me know, I shall attach them.

Also, the above instruction has me confused a bit: "you can install them" meaning you have shared instructions to install firedbg on my pi4? or you wish to add these instructions to the local installation guide?

**Reply by tyt2y3** [reply]
Sorry for the confusion, we just updated the BUILD.rs https://github.com/SeaQL/FireDBG.for.Rust/commit/e3bc7549899145046534e17e0b0d8a99ba77de28
